napari/_qt/widgets/qt_viewer_buttons.py

```diff
@@ -183,10 +183,11 @@ def open_perspective_popup(self):
         # make slider connected to perspective parameter
         sld = QSlider(Qt.Orientation.Horizontal, self)
         sld.setRange(0, max(90, int(self.viewer.camera.perspective)))
-        sld.setValue(self.viewer.camera.perspective)
+        sld.setValue(int(self.viewer.camera.perspective))
         sld.valueChanged.connect(
             lambda v: setattr(self.viewer.camera, 'perspective', v)
         )
+        self.perspective_slider = sld
```

Comment:
This is for the tests right?

Reply:
Yes! This enables doing the test in a simpler manner. However, I also added this here to keep some consistency with the way the grid mode popup is implemented (there also a reference to the widgets from the popup is created):

https://github.com/napari/napari/blob/87108ea1cb11501f5659b9d97a80c27a61a6ce32/napari/_qt/widgets/qt_viewer_buttons.py#L238-L272